include/boost/geometry/algorithms/detail/buffer/buffer_inserter.hpp

```diff
@@ -175,6 +176,16 @@ struct buffer_range
         }
     }
 
+    static inline bool similar_direction(output_point_type const& p0,
+            output_point_type const& p1,
+            output_point_type const& p2)
+    {
+        typedef arithmetic::general_form<coordinate_type> gf;
+        gf const p = arithmetic::construct_line<coordinate_type>(p0, p1);
+        gf const q = arithmetic::construct_line<coordinate_type>(p1, p2);
```

Comment:
Why not simply use a constructor?

Reply:
See my other answer, it is concept based. I can rename it to `make` which is more common.